### PR TITLE
HighlightInfo: unknownUnlock is Highlight Type

### PR DESCRIPTION
**Pull request by neptuwunium**
The C2 files it links to are simple STU pointers to a 07C prostring.

### CODE REVIEW

File: OverwatchHighlights/Extensions.cs [hunk 1]
```diff
@@ -62,6 +62,16 @@ public static uint PeekUInt32(this BinaryReader br)
 			return value;
 		}
 
+		public static HighlightType ReadHighlightType64(this BinaryReader br)
+		{
+			const ulong HighlightTypeMask = 0x0830000000000000ul;
+			ulong value = br.ReadUInt64();
+			Debug.Assert((value & 0xFFFFFFFF00000000ul) == HighlightTypeMask);
```
Comment (lunasorcery) [inline OverwatchHighlights/Extensions.cs]: When the value is zero, the mask bits aren't set, so this check ought to be `(value & 0xFFFFFFFF00000000ul) == HighlightTypeMask || value == 0`

File: OverwatchHighlights/HighlightInfo.cs [hunk 2]
```diff
@@ -98,7 +85,7 @@ public void Print()
 			Console.WriteLine($"  Skin: {skin}");
 			Console.WriteLine($"  Weapon: {weaponSkin}");
 			Console.WriteLine($"  Intro: {highlightIntro}");
-			Console.WriteLine($"  Unknown Unlock: {unknownUnlock:X16}");
+			Console.WriteLine($"  Type: {highlightType:X16}");
```
Comment (lunasorcery) [inline OverwatchHighlights/HighlightInfo.cs]: The `X16` formatter won't work on an enum type, you should be safe to just put `{highlightType}` here.

File: OverwatchHighlights/HighlightTypes.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace OverwatchHighlights
+{
+    public enum HighlightType : uint
+    {
+        SHUTDOWN = 0x00000001,
+        SHARPSHOOTER = 0x00000002,
+        BLANK = 0x00000003,
+        LIFESAVER = 0x00000004
```
Comment (lunasorcery) [inline OverwatchHighlights/HighlightTypes.cs]: I'd prefer that these not be uppercase, and be space-aligned to keep stylistic consistency with the rest of the codebase. Also needs an entry for zero.

**Review by lunasorcery — CHANGES_REQUESTED** [review]
Looks good, just fix up the few things I've pointed out and I'll be happy to merge :)

**Comment by neptuwunium** [comment]
That should cover it.

**Review by lunasorcery — APPROVED** [review]

**Pull request merged**